### PR TITLE
Fix deprecation warning by upgrading rspec-rails and shoulda-matchers

### PR DESCRIPTION
**Pull request by monal09**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,14 +6,15 @@ gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: '3-0-stable'
...
+group :assets do
   gem 'coffee-rails', '4.0.1'
 end
 
 group :test do
-  gem 'rspec-rails', '~> 2.10'
-  gem 'shoulda-matchers', '2.2.0'
+  gem 'rspec-rails', '~> 3.5.2'
+  gem 'shoulda-matchers', '3.1.1'
   gem 'simplecov', :require => false
```
Comment (tanmay3011) [inline Gemfile]: @monal09 Please fix all these old ruby rocket operator syntax

File: Gemfile [hunk 2]
```diff
@@ -6,14 +6,15 @@ gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: '3-0-stable'
...
   gem 'database_cleaner'
```
Comment (tanmay3011) [inline Gemfile]: @monal09 lock all versions

File: spec/controllers/spree/admin/favorite_products_controller_spec.rb [hunk 3]
```diff
@@ -29,50 +29,50 @@
 
   describe "GET index" do
     def send_request
-      get :index, :page => 1 ,:use_route => 'spree', :q => { 's' => 'name desc' }
+      spree_get :index, page: 1, :q => { 's' => 'name desc' }
```
Comment (tanmay3011) [inline spec/controllers/spree/admin/favorite_products_controller_spec.rb]: @monal09 Please fix old ruby syntax

File: app/models/spree/favorite.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
 module Spree
   class Favorite < ActiveRecord::Base
-    belongs_to :product
-    belongs_to :user
-    validates :user_id, :product_id, presence: true
-    validates :product_id, uniqueness: { scope: :user_id, message: "already marked as favorite" }
-    validates :product, presence: { message: "is invalid" }, if: :product_id
+    belongs_to :product, required: true
```
Comment (tanmay3011) [inline app/models/spree/favorite.rb]: @monal09 Use with_options

File: app/models/spree/product_decorator.rb [hunk 5]
```diff
@@ -3,10 +3,10 @@
   has_many :favorite_users, through: :favorites, class_name: 'Spree::User', source: 'user'
 
   def self.favorite
-    joins(:favorites).group('spree_favorites.product_id')
+    joins(:favorites).group('product_id')
```
Comment (tanmay3011) [inline app/models/spree/product_decorator.rb]: @monal09 why string? why not symbol?

File: spec/models/spree/favorite_spec.rb [hunk 6]
```diff
@@ -24,7 +24,7 @@
 
     it "does not validate the presence of product" do
       @favorite.valid?
-      expect(@favorite.errors[:product]).to eq([])
+      expect(@favorite.errors[:product]).to eq(["can't be blank"])
```
Comment (tanmay3011) [inline spec/models/spree/favorite_spec.rb]: @monal09 remove these context spec as shoulda is already testing presence

File: app/models/spree/favorite.rb [hunk 7]
```diff
@@ -1,7 +1,10 @@
 module Spree
   class Favorite < ActiveRecord::Base
-    belongs_to :product, required: true
-    belongs_to :user, required: true
+
+  with_options required: true do |favorite|
```
Comment (tanmay3011) [inline app/models/spree/favorite.rb]: @monal09 We don't need to have an extra variable
`with_options required: true do
  belongs_to :product
  belongs_to :user
end`

File: app/controllers/spree/favorite_products_controller.rb [hunk 8]
```diff
@@ -2,16 +2,16 @@ module Spree
   class FavoriteProductsController < Spree::StoreController
 
     before_filter :authenticate_spree_user!
-    before_filter :find_favorite_product, :only => :destroy
+    before_filter :find_favorite_product, only: :destroy
```
Comment (nishant-cyro) [inline app/controllers/spree/favorite_products_controller.rb]: Use before_action.

File: app/controllers/spree/favorite_products_controller.rb [hunk 9]
```diff
@@ -28,7 +28,7 @@ def destroy
 
     private
       def find_favorite_product
-        @favorite = spree_current_user.favorites.joins(:product).readonly(false).where(:spree_products => {:id => params[:id]}).first
+        @favorite = spree_current_user.favorites.joins(:product).readonly(false).where(spree_products: {id: params[:id]}).first
```
Comment (nishant-cyro) [inline app/controllers/spree/favorite_products_controller.rb]: Use merge here.

File: app/models/spree/product_decorator.rb [hunk 10]
```diff
@@ -1,12 +1,12 @@
 Spree::Product.class_eval do
-  has_many :favorites, :dependent => :destroy
-  has_many :favorite_users, :through => :favorites, :class_name => 'Spree::User', :source => 'user'
+  has_many :favorites, dependent: :destroy
+  has_many :favorite_users, through: :favorites, class_name: 'Spree::User', source: 'user'
 
   def self.favorite
-    joins(:favorites).group('spree_favorites.product_id')
+    joins(:favorites).group(:product_id)
```
Comment (nishant-cyro) [inline app/models/spree/product_decorator.rb]: Please have a look at this.

**Pull request merged**